### PR TITLE
Allow editing assigned team via public api

### PR DESCRIPTION
**Pull request by iskhakov**
# What this PR does

For example, changing the integration's team
```
curl "localhost:8080/api/v1/integrations/CLYV3QBVHDV3T/" \
  --request PUT \
  --header "Authorization: meow" \
  --header "Content-Type: application/json" --data '{"team_id": "TWP6JJJN6LYZX"}'
```

## Checklist

- [ ] Unit, integration, and e2e (if applicable) tests updated
- [ ] Documentation added (or `pr:no public docs` PR label added if not required)
- [ ] `CHANGELOG.md` updated (or `pr:no changelog` PR label added if not required)

### CODE REVIEW

File: engine/apps/public_api/serializers/schedules_web.py [hunk 1]
```diff
@@ -11,6 +11,7 @@
 
 
 class ScheduleWebSerializer(ScheduleBaseSerializer):
+    team_id = TeamPrimaryKeyRelatedField(required=False, allow_null=True, source="team")
```
Comment (Konstantinov-Innokentii) [inline engine/apps/public_api/serializers/schedules_web.py:14]: there is  
```team = TeamPrimaryKeyRelatedField(allow_null=True, default=CurrentTeamDefault())``` in ScheduleBaseSerializer

File: engine/apps/public_api/views/escalation_chains.py [hunk 2]
```diff
@@ -22,7 +21,7 @@ class EscalationChainView(RateLimitHeadersMixin, UpdateSerializerMixin, ModelVie
 
     model = EscalationChain
     serializer_class = EscalationChainSerializer
-    update_serializer_class = EscalationChainUpdateSerializer
+    update_serializer_class = EscalationChainSerializer
```
Comment (Konstantinov-Innokentii) [inline engine/apps/public_api/views/escalation_chains.py:24]: You can just remove UpdateSerializerMixin and update_serializer_class from EscalationChainView and View will always serializer_class

File: engine/apps/public_api/serializers/schedules_calendar.py [hunk 3]
```diff
@@ -11,6 +11,7 @@
 
 
 class ScheduleCalendarSerializer(ScheduleBaseSerializer):
+    team_id = TeamPrimaryKeyRelatedField(required=False, allow_null=True, source="team")
```
Comment (Konstantinov-Innokentii) [inline engine/apps/public_api/serializers/schedules_calendar.py:14]: There is team_id field in  ScheduleBaseSerializer, is second one necessary?

File: engine/apps/public_api/views/escalation_chains.py (hunk 2 above)
Comment (iskhakov) [reply engine/apps/public_api/views/escalation_chains.py:24]:
```suggestion
```

File: engine/apps/public_api/serializers/schedules_calendar.py (hunk 3 above)
Comment (iskhakov) [reply engine/apps/public_api/serializers/schedules_calendar.py:14]:
```suggestion
```

**Comment by Konstantinov-Innokentii** [comment]
LGTM, left some comments. 
At that moment, terraform (as well as api-client) doesn't support team_id parameter for updating resources

**Comment by iskhakov** [comment]
Changed api client and terraform plugin
https://github.com/grafana/terraform-provider-grafana/pull/864
https://github.com/grafana/amixr-api-go-client/pull/11

**Review by Konstantinov-Innokentii — APPROVED** [review]

**Pull request merged**